test/e2e/traceflow_test.go

```diff
@@ -0,0 +1,330 @@
+	}
+}
+
+func (data *TestData) enableTraceflow(t *testing.T) error {
+	configMap, err := data.GetAntreaConfigMap(antreaNamespace)
+	if err != nil {
+		t.Fatalf("Failed to get ConfigMap: %v", err)
+	}
+
+	// Enable Traceflow in antrea-controller and antrea-agent ConfigMap.
+	// Use Geneve tunnel.
+	antreaControllerConf, _ := configMap.Data["antrea-controller.conf"]
+	antreaControllerConf = strings.Replace(antreaControllerConf, "#  Traceflow: false", " Traceflow: true", 1)
+	configMap.Data["antrea-controller.conf"] = antreaControllerConf
+	antreaAgentConf, _ := configMap.Data["antrea-agent.conf"]
+	antreaAgentConf = strings.Replace(antreaAgentConf, "#  Traceflow: false", " Traceflow: true", 1)
+	antreaAgentConf = strings.Replace(antreaAgentConf, "#tunnelType: geneve", "tunnelType: geneve", 1)
+	configMap.Data["antrea-agent.conf"] = antreaAgentConf
+
+	if _, err := data.clientset.CoreV1().ConfigMaps(antreaNamespace).Update(context.TODO(), configMap, metav1.UpdateOptions{}); err != nil {
+		return fmt.Errorf("failed to update ConfigMap %s: %v", configMap.Name, err)
+	}
+	_, err = data.restartAntreaControllerPod(defaultTimeout)
+	if err != nil {
+		return fmt.Errorf("error when restarting antrea-controller Pod: %v", err)
+	}
+	_, err = data.restartAntreaAgentPod(defaultTimeout)
+	if err != nil {
+		return fmt.Errorf("error when restarting antrea-agent Pod: %v", err)
+	}
+
+	return nil
+}
+
+func fillTraceflow(src string, dest string) *v1alpha1.Traceflow {
+	return &v1alpha1.Traceflow{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: randName("test-traceflow-"),
+		},
+		Spec: v1alpha1.TraceflowSpec{
+			Source: v1alpha1.Source{
+				Namespace: testNamespace,
+				Pod:       src,
+			},
+			Destination: v1alpha1.Destination{
+				Namespace: testNamespace,
+				Pod:       dest,
+			},
+			Packet: v1alpha1.Packet{
+				IPHeader: v1alpha1.IPHeader{
+					Protocol: 6,
+				},
+				TransportHeader: v1alpha1.TransportHeader{
+					TCP: &v1alpha1.TCPHeader{
+						DstPort: 80,
+					},
+				},
+			},
+		},
+	}
+}
+
+// compareObservations compares expected results and actual results.
+func compareObservations(expected v1alpha1.NodeResult, actual v1alpha1.NodeResult, t *testing.T) {
+	if expected.Node != actual.Node {
+		t.Errorf("Error: NodeResult should be on %s, but is on %s", expected.Node, actual.Node)
+	}
+	exObs := expected.Observations
+	acObs := actual.Observations
+	if len(exObs) != len(acObs) {
+		t.Errorf("Error: Observations should be %v, but got %v", exObs, acObs)
+	}
+	for i := 0; i < len(exObs); i++ {
+		if exObs[i].Component != acObs[i].Component ||
+			exObs[i].ComponentInfo != acObs[i].ComponentInfo ||
+			exObs[i].Action != acObs[i].Action {
+			t.Errorf("Error: Observations should be %v, but got %v", exObs, acObs)
+		}
+	}
+}
+
+// createNPDenyAllIngress creates a NetworkPolicy that denies ingress traffic of specific label.
+func (data *TestData) createNPDenyIngress(key string, value string, name string) (*networkingv1.NetworkPolicy, error) {
```

Comment:
```suggestion
func (data *TestData) createNPDenyAllIngress(key string, value string, name string) (*networkingv1.NetworkPolicy, error) {
```
to align with your comment and the `createNPAllowAllEgress`

Reply:
But it doesn't deny all ingress. It uses a label.

Reply:
No, it denies all ingress traffic but applies to pods with specific label. Comment is also wrong.

Reply:
You are right. I though "all" refers to "all pods".